### PR TITLE
fix: bun-2872 change api

### PR DESCRIPTION
**Pull request by BrianJiang2021**
Jira: [BUN-2872](https://bigc-b2b.atlassian.net/browse/BUN-2872)

## What/Why?
Convert all Open Source Buyer Portal v2/bc endpoints to GraphQL mutations/queries

## Rollout/Rollback
undo pr

## Testing
87a24024f7e051ef76d7c6dfd88c0785

925632b2404b6ec874012b1d7ef215d7

https://github.com/user-attachments/assets/4bbe778c-cb48-4938-bdd2-c93936f22bd4

https://github.com/user-attachments/assets/2acf96b0-baaf-4e4b-8053-bfbe09e47bb0

### CODE REVIEW

File: apps/storefront/src/shared/service/b2b/graphql/global.ts [hunk 1]
```diff
@@ -193,6 +194,81 @@ const companyCreditConfig = () => `{
   }
 }`;
 
+const PricingProductItemOptions = `
+	options {
+		optionId,
+		valueId,
+	}
```
Comment (icatalina) [inline apps/storefront/src/shared/service/b2b/graphql/global.ts:201]: tabs?

File: apps/storefront/src/types/products.ts [hunk 2]
```diff
@@ -209,3 +209,25 @@ export interface CalculatedProducts {
...
+
+export interface BulkPriceItem {
+  minimum: number;
+  maximum: number;
+  discountAmount: number;
+  discount_amount?: number;
+  discountType: string;
+  discount_type?: string;
+  taxDiscountAmount: TaxDiscountAmountProps;
+  tax_discount_amount?: TaxDiscountAmountProps;
```
Comment (icatalina) [inline apps/storefront/src/types/products.ts:232]: Do these two contain all the possible combinations of keys? or should they be:

```tsx
taxDiscountAmount: {
  asEntered: number;
  enteredInclusive: boolean;
  // ...
},
tax_discount_amount?: {
  as_entered: number;
  entered_inclusive: boolean;
  // ...
}
```

File: apps/storefront/src/utils/loginInfo.ts [hunk 3]
```diff
@@ -112,7 +112,7 @@ export const getloginTokenInfo = () => {
 export const loginInfo = async () => {
   const loginTokenInfo = getloginTokenInfo();
   const {
-    data: { token },
+    storeFrontToken: { token },
```
Comment (icatalina) [inline apps/storefront/src/utils/loginInfo.ts:115]: As with previous changes like this, we might want to absorb this destructuring into the `getBCGraphqlToken` method:

```tsx
const token = await getBCGraphqlToken(loginTokenInfo);
```

File: apps/storefront/src/types/products.ts [hunk 4]
```diff
@@ -209,3 +209,25 @@ export interface CalculatedProducts {
...
+  tax_inclusive?: number;
+}
+
+export interface BulkPriceItem {
+  minimum: number;
+  maximum: number;
+  discountAmount: number;
+  discount_amount?: number;
+  discountType: string;
+  discount_type?: string;
```
Comment (icatalina) [inline apps/storefront/src/types/products.ts:230]: Having two possible representation of each key is a bit awkward. This is normally better solved with an anti corruption layer, basically transforming all the data at the edge of the application so the interfaces are consistent inside the app.

File: apps/storefront/src/shared/service/b2b/graphql/register.ts [hunk 5]
```diff
@@ -225,3 +410,13 @@ export const getBCStoreChannelId = () =>
     query: getStoreChannelId,
     variables: { storeHash, bcChannelId: channelId },
   });
+
+export const createBCCompanyUser = (data: Partial<CreateCustomer>) =>
```
Comment (icatalina) [inline apps/storefront/src/shared/service/b2b/graphql/register.ts:414]: 🤔 are all the properties of the `CreateCustomer` optional? I'd imagine some (`email`?) might be required.

File: apps/storefront/src/shared/service/b2b/graphql/register.ts [hunk 6]
```diff
@@ -225,3 +410,13 @@ export const getBCStoreChannelId = () =>
...
+  B3Request.graphqlB2B({
+    query: customerCreate(data),
+  });
+
+export const sendSubscribersState = (data: Partial<CustomerSubscribers>) =>
```
Comment (icatalina) [inline apps/storefront/src/shared/service/b2b/graphql/register.ts:419]: Same as above, are all the properties optional here?

File: apps/storefront/src/shared/service/b2b/graphql/login.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import B3Request from '../../request/b3Fetch';
+
+interface ApiTokenConfig {
+  storeHash: string;
+  channel_id: number;
+  expires_at: number;
+  allowed_cors_origins: string[];
+}
+
+const storeFrontToken = (data: Partial<ApiTokenConfig>) => `mutation {
```
Comment (icatalina) [inline apps/storefront/src/shared/service/b2b/graphql/login.ts:12]: Are all the properties in ApiTokenConfig optional when calling the `storeFrontToken` method?

**Comment by BrianJiang2021** [comment]
The content of this PR has been covered by [PR: https://github.com/bigcommerce/b2b-buyer-portal/pull/128]. So this PR can be closed.

**Pull request closed without merging**